### PR TITLE
update docs for sso to address provider issue

### PR DESCRIPTION
**Pull request by novekm**
### Description
This PR addresses the bug with `aws_ssoadmin_account_assignment` and `aws_ssoadmin_managed_policy_attachment` listed in this [GitHub Issue - #33337](https://github.com/hashicorp/terraform-provider-aws/issues/33337). Per @jar-b's further investigation, adding an explicit dependency between these two resources resolves a provider issue where both resources are attempted to be destroyed at the same time, causing `terraform destroy` to fail until the command is run a second time.

This PR adds clear documentation of this error and the resolution to resource documentation pages for the above affected resources.

### Relations

Closes [#33337](https://github.com/hashicorp/terraform-provider-aws/issues/33337)
Closes[#33384](https://github.com/hashicorp/terraform-provider-aws/pull/33384)

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccXXX PKG=ec2

...
```

### CODE REVIEW

File: website/docs/r/ssoadmin_managed_policy_attachment.html.markdown [hunk 1]
```diff
@@ -10,6 +10,9 @@ description: |-
 
 Provides an IAM managed policy for a Single Sign-On (SSO) Permission Set resource
 
+~> **IMPORTANT:** Currently there is an issue where deletion of both an `aws_ssoadmin_managed_policy_attachment` and `aws_ssoadmin_account_assignment` can occur simultaneously. This causes a problem with the `Delete` operation of the policy attachment and can cause your `terraform destroy` to fail. To avoid this, add an explicit dependency between these resources with the `depends_on` meta argument.
```
Comment (jar-b) [inline website/docs/r/ssoadmin_managed_policy_attachment.html.markdown:13]: I'm not sure this is a bug so much as a nuance of the relationship between these particular resources during destruction. Perhaps adding an example on the managed policy attachment page that includes the `depends_on` argument along with this context could provide more directed guidance?

```hcl
resource "aws_ssoadmin_managed_policy_attachment" "example" {
  # Adding an explicit dependency on the account assignment resource will
  # allow the managed attachment to be safely destroyed prior to the removal
  # of the account assignment.
  depends_on = [aws_ssoadmin_account_assignment.example]

  instance_arn       = tolist(data.aws_ssoadmin_instances.test.arns)[0]
  managed_policy_arn = "arn:aws:iam::aws:policy/my-policy"
  permission_set_arn = aws_ssoadmin_permission_set.example.arn
}
```

Comment (novekm) [reply website/docs/r/ssoadmin_managed_policy_attachment.html.markdown]: just updated examples for docs for both resources to include this 👍🏾

**Review by jar-b — APPROVED** [review]
LGTM 🎉

**Comment by jar-b** [comment]
Thanks for your contribution, @novekm! 👍

**Pull request merged**